evaluation/evaluate.py

```
@@ -659,6 +681,90 @@ def evaluate_one(module, item, index, judge_llm=None):
+
+        data = {
+            "user_input":        [r["question"]         for _, r in valid],
+            "response":          [r[answer_key]          for _, r in valid],
+            "retrieved_contexts":[r.get(ctx_key, [])     for _, r in valid],
+            "reference":         [r.get("ground_truth", "") for _, r in valid],
+        }
+        try:
+            dataset = HFDataset.from_dict(data)
+            result = ragas_evaluate(dataset, metrics=metrics)
+            result_df = result.to_pandas()
+            for df_idx, (row_idx, row) in enumerate(valid):
```

Ruff (0.15.15)

[warning] Loop control variable `row_idx` not used within loop body

Rename unused `row_idx` to `_row_idx`

(B007)